app/src/main/kotlin/com/github/gotify/messages/provider/MessageWithImage.kt

```diff
@@ -2,7 +2,13 @@ package com.github.gotify.messages.provider
 
 import com.github.gotify.client.model.Message
 
+@Suppress("EqualsOrHashCode")
 internal data class MessageWithImage(
     val message: Message,
     val image: String?
-)
+) {
+    override fun equals(other: Any?): Boolean {
```

Comment:
Why is this needed. I'd expect the Kotlin data class equals to be the same. If it's needed, then please also implement hashcode, so it works the same as equals.

Reply:
You are right, it indeed does work now without custom logic.  
I've had some issues with the DiffUtil not recognizing differences, that's probably why I've implemented it.